c10/core/MemoryFormat.h

```diff
@@ -127,7 +127,7 @@ inline bool is_channels_last_strides_2d_s4(
     if (sizes[d] == 0) {
       return false;
     }
-    if (strides[d] < min) {
+    if (strides[d] < min && !(d == 0 && sizes[0] == 1)) {
```

Comment:
Thank you for pointing me out. I added a comment here.

Reply:
I would suggest you explain the essential reason instead of making others feel you are working around an issue. Why does reshape return unexpected stride that would cause the issue? Why can't we fix the problem there?

Reply:
Reshape return unexpected stride: The root cause is that reshape is hard to handle the ambiguous strides of non-contiguous tensors and channels_last tensors with size=1 for dims. 

For example: Reshape the channels_last tensor  [1, 2, 3, 4]@[24, 1, 8, 2] to the shape [1, 1, 2, 1, 2, 3]. Reshape separate old shape into chunks of dimensions, where the dimensions are contiguous in each chunk. [1, 2, 3, 4] will be chunked into [4, 3], [2, 1] in inverse order. New shape must be able to be separated into same number of chunks as old shape was separated into, i.e. [3, 2, 1], [2, 1, 1] where each chunk of new shape has matching numel as the corresponding chunk of old shape and each chunk's minimum stride of new stride is the same as that of old stride. 

Note that It is hard to decide which chunk `1` should belong to. For the new shape [2, 1, 1]  ? [2, 1], [1] ?  or [2], [1], [1] ? Therefore reshape will put `1` to the previous chunk by default, and compute the new stride [2, 2, 1, 12, 6, 2]. 

Similarly, in the real scenario: reshape tensor [1, 2, 3, 4]@[24, 1, 8, 2] to the shape [1, 2, 3, 4]. New shape is separated into [4, 3], [2, 1] and get the new stride [2, 1, 8, 2].